src/rpc/methods/state.rs

```diff
@@ -100,37 +101,53 @@ pub const MSIG_GET_PENDING: &str = "Filecoin.MsigGetPending";
 pub const STATE_MINER_SECTORS: &str = "Filecoin.StateMinerSectors";
 pub const STATE_MINER_PARTITIONS: &str = "Filecoin.StateMinerPartitions";
 
-pub async fn miner_get_base_info<DB: Blockstore + Send + Sync + 'static>(
-    params: Params<'_>,
-    data: Ctx<DB>,
-) -> anyhow::Result<LotusJson<Option<MiningBaseInfo>>, ServerError> {
-    let LotusJson((address, epoch, ApiTipsetKey(tsk))) = params.parse()?;
+pub enum MinerGetBaseInfo {}
+impl RpcMethod<3> for MinerGetBaseInfo {
+    const NAME: &'static str = "Filecoin.MinerGetBaseInfo";
+    const PARAM_NAMES: [&'static str; 3] = ["address", "epoch", "tsk"];
+    const API_VERSION: ApiVersion = ApiVersion::V0;
 
-    let ts = data
-        .state_manager
-        .chain_store()
-        .load_required_tipset_or_heaviest(&tsk)?;
+    type Params = (LotusJson<Address>, i64, LotusJson<ApiTipsetKey>);
+    type Ok = Option<MiningBaseInfo>;
 
-    data.state_manager
-        .miner_get_base_info(data.state_manager.beacon_schedule(), ts, address, epoch)
-        .await
-        .map(|info| Ok(LotusJson(info)))?
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (LotusJson(address), epoch, LotusJson(ApiTipsetKey(tsk))): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let ts = ctx
+            .state_manager
+            .chain_store()
+            .load_required_tipset_or_heaviest(&tsk)?;
+
+        Ok(ctx
+            .state_manager
+            .miner_get_base_info(ctx.state_manager.beacon_schedule(), ts, address, epoch)
+            .await?)
+    }
 }
-/// runs the given message and returns its result without any persisted changes.
-pub async fn state_call<DB: Blockstore + Send + Sync + 'static>(
-    params: Params<'_>,
-    data: Ctx<DB>,
-) -> Result<ApiInvocResult, ServerError> {
-    let LotusJson((message, ApiTipsetKey(key))) = params.parse()?;
 
-    let state_manager = &data.state_manager;
-    let tipset = data
-        .state_manager
-        .chain_store()
-        .load_required_tipset_or_heaviest(&key)?;
-    // Handle expensive fork error?
-    // TODO(elmattic): https://github.com/ChainSafe/forest/issues/3733
-    Ok(state_manager.call(&message, Some(tipset))?)
+pub enum StateCall {}
+impl RpcMethod<2> for StateCall {
+    const NAME: &'static str = "Filecoin.StateCall";
+    const PARAM_NAMES: [&'static str; 2] = ["message", "tsk"];
+    const API_VERSION: ApiVersion = ApiVersion::V0;
+
+    type Params = (LotusJson<Message>, LotusJson<ApiTipsetKey>);
+    type Ok = ApiInvocResult;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (LotusJson(message), LotusJson(ApiTipsetKey(tsk))): Self::Params,
```

Comment:
Hope we can get rid of `LotusJson` wrappers in `Params` as well

Reply:
> Hope we can get rid of `LotusJson` wrappers in `Params` as well

Yes, exactly what I was thinking. It's quite verbose both here and in `api compare` tests.

Reply:
@aatifsyed I made a draft PR to drop these wrappers, we can use it for discussion.